threedi_modelchecker/checks/other.py

```diff
@@ -504,3 +504,38 @@ def get_position(point, linestring):
 
     def description(self) -> str:
         return f"{self.column_name} must be more than {self.min_distance} m apart (or exactly on the same position)"
+
+
+class PumpStorageTimestepCheck(BaseCheck):
+    """Check that a pumpstation will not empty its storage area within one timestep"""
+
+    def get_invalid(self, session: Session) -> List[NamedTuple]:
+        return (
+            session.query(models.Pumpstation)
+            .join(
+                models.ConnectionNode,
+                models.Pumpstation.connection_node_start_id == models.ConnectionNode.id,
+            )
+            .filter(
+                (models.ConnectionNode.storage_area != None)
+                & (
+                    (  # calculate how many seconds the pumpstation takes to empty its storage: (storage * height)/pump capacity
+                        (
+                            models.ConnectionNode.storage_area
+                            * (
+                                models.Pumpstation.start_level
+                                - models.Pumpstation.lower_stop_level
+                            )
+                        )
+                        / models.Pumpstation.capacity
```

Comment:
Probably better to switch the capacity to the other side of the `<` to avoid zero division.

So you compare `storage_area * (start_level) - lower_stop_level` against `sim_time_step * capacity`.